### PR TITLE
fix: upload url path and add logging in the SDK

### PR DESCRIPTION
**Pull request by Natoandro**
- Fix upload url: prepare-upload returns upload tokens instead of upload urls
- Add logging in the typegraph SDK
- Refactor the actor system in the CLI
- Use jsonrpc for communication between the CLI and typegraph processes (over stdin/stdout)

#### Migration notes

- The `typegraphs.deno` section of the `metatype.yaml` config file has been replaced by `typegraphs.typescript` and `typegraphs.javascript`.
- `tg_deploy` params has changed.

- [x] The change comes with new or modified tests
- [x] Hard-to-understand functions have explanatory comments
- [x] End-user documentation is updated to reflect the change

### CODE REVIEW

**Review by Yohe-Am — APPROVED** [review]
Heroic work. Will hopefully be the last major refactor this quarter lol

File: typegraph/node/sdk/src/io.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      jsonrpc: JSONRPC_VERSION,
+      id: rpcId,
+      method,
+      params,
+    });
+
+    process.stdout.write(`jsonrpc: ${rpcMessage}\n`);
+    return responseReader.read(rpcId);
+  };
+})();
```
Comment (afmika) [inline typegraph/node/sdk/src/io.ts:110]: Curious though, what happens when `rpcMessage` contains the endline character?

**Review by afmika — COMMENTED** [review]
I guess reverting back to stdout but with a better protocol on top is also worth trying considering the server approach had some overhead and rise in maintenance difficulty. 👍

File: typegraph/node/sdk/src/io.ts (hunk 1 above)
Comment (Natoandro) [reply typegraph/node/sdk/src/io.ts:110]: `JSON.strigify` would never emit an endline character...

**Review by zifeo — COMMENTED** [review]
Bravo!

File: meta-cli/src/com/store.rs [hunk 2]
```diff
@@ -7,6 +7,7 @@ use common::node::BasicAuth;
 use lazy_static::lazy_static;
 use serde::{Deserialize, Serialize};
 use std::borrow::{Borrow, BorrowMut};
+use std::sync::Mutex;
```
[the remark this answers is not in the document]
Comment (Natoandro) [reply meta-cli/src/com/store.rs:10]: Not related to the actor system. Anyway, this file will be removed...

File: meta-cli/src/deploy/actors/task.rs [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+use message::*;
+
+#[derive(Serialize, Deserialize)]
+#[serde(untagged)]
+enum TaskOutput {
+    Serialized(Typegraph),
+    Deployed { deployed: String },
+}
+
+// TODO cli param
```
[the remark this answers is not in the document]
Comment (Natoandro) [reply meta-cli/src/deploy/actors/task.rs:77]: That was just a question a was asking myself. But now I think it is better configured with env variables...

File: libs/metagen/metatype.yaml
[the remark this answers is not in the document]
Comment (Natoandro) [reply libs/metagen/metatype.yaml:1]: The migration files are supposed to be relative to the parent directory of the config file. So it is required unless we change the way they are resolved...

File: typegate/tests/metagen/__snapshots__/metagen_test.ts.snap
[the remark this answers is not in the document]
Comment (Yohe-Am) [reply typegate/tests/metagen/__snapshots__/metagen_test.ts.snap:1]: Nah, it shouldn't be empty.

**Review by Yohe-Am — APPROVED** [review]

**Pull request merged**